CMake/AbseilHelpers.cmake

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2017 The Abseil Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+function(abseil_library Name)
```

Comment:
These functions should probably be documented.

Reply:
Additionally these functions should use `cmake_parse_arguments` to accept various inputs instead of global names based on the input `Name`. Here is what I rigged up https://github.com/efcs/abseil-cpp/blob/with-cmake/CMake/AbseilHelpers.cmake (Example usage: https://github.com/efcs/abseil-cpp/blob/with-cmake/absl/strings/CMakeLists.txt). 

@adevress feel free to steal from my patches.

Reply:
> These functions should probably be documented.

Yes, just let me time to do it, you commented on that less than 2h after I commited them :)

> Additionally these functions should use cmake_parse_arguments to accept various inputs instead of global names based on the input Name

I know about cmake_parse_arguments. However I do not really appreciate it for this usage. It makes quickly cmake function calls very verbose and with 32 parameters, reusing variables that have been declared right before. I tend to think that adopting a simple naming / namespace convention instead of manual calls make the entire thing simpler and more readable.

Reply:
> I know about cmake_parse_arguments. However I do not really appreciate it for this usage. It makes quickly cmake function calls very verbose and with 32 parameters, reusing variables that have been declared right before.

Not to be rude, but using variables is fundamentally no better, and in fact worse, than passing named arguments using `cmake_parse_arguments`. You can easily have 32 global variables, for example.

> I tend to think that adopting a simple naming / namespace convention instead of manual calls make the entire thing simpler and more readable.

You can adopt a simple naming scheme for the argument keywords as well. But using global variables instead of function arguments *does not* make things easier to read; it does the opposite. I have no idea where or if the variable is even being used.

We should be using idiomatic CMake. I don't think you're current formulation is OK.

Reply:
Done. Every function has been refactored to a version with explicit parameters.